### PR TITLE
[0.5.0] Live update on chart revision number

### PR DESCRIPTION
**Pull request by jnfrati**
## Pull request type
Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Other (please describe):

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] If it's a backend change, tests for the changes have been added and `go test ./...` runs successfully from the root folder.
- [x] If it's a frontend change, Prettier has been run
- [ ] Docs have been reviewed and added / updated if needed

## What is the current behavior?

Revision number is not updating when webhook is triggered for a chart. 

## What is the new behavior?

Revision number will perform live updates when it changes

## Technical Spec/Implementation Notes

Implement `factory.Core().V1().Secrets().Informer()` to get updates on secrets with websockets, with all the updates filter them to get only the last helm.sh/release.v1 object for each of the charts displayed

### CODE REVIEW

**Review by abelanger5 — CHANGES_REQUESTED** [review]
Hi @jnfrati, just tested and it's mostly working as expected. There are a few issues:
- The websocket termination is not handled gracefully from the backend. I'm not sure if this was due to a change that you introduced, but it crashes the entire backend. Here is the error:

```go
porter_1       | E0625 12:28:13.663557     241 runtime.go:78] Observed a panic: "send on closed channel" (send on closed channel)
porter_1       | goroutine 2814 [running]:
porter_1       | k8s.io/apimachinery/pkg/util/runtime.logPanic(0x237c1e0, 0x29fd1a0)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/runtime/runtime.go:74 +0xa6
porter_1       | k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/runtime/runtime.go:48 +0x89
porter_1       | panic(0x237c1e0, 0x29fd1a0)
porter_1       |        /usr/local/go/src/runtime/panic.go:969 +0x1b9
porter_1       | github.com/porter-dev/porter/internal/kubernetes.(*Agent).StreamControllerStatus.func3(0x2696880, 0xc006c0af00)
porter_1       |        /porter/internal/kubernetes/agent.go:592 +0x13d
porter_1       | k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
porter_1       |        /go/pkg/mod/k8s.io/client-go@v0.18.8/tools/cache/controller.go:218
porter_1       | k8s.io/client-go/tools/cache.(*processorListener).run.func1()
porter_1       |        /go/pkg/mod/k8s.io/client-go@v0.18.8/tools/cache/shared_informer.go:744 +0xc2
porter_1       | k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc001b8c760)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:155 +0x5f
porter_1       | k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc003a0df60, 0x2a38ec0, 0xc000951200, 0x231f801, 0xc00384c1e0)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:156 +0xad
porter_1       | k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc001b8c760, 0x3b9aca00, 0x0, 0x474601, 0xc00384c1e0)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:133 +0x98
porter_1       | k8s.io/apimachinery/pkg/util/wait.Until(...)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:90
porter_1       | k8s.io/client-go/tools/cache.(*processorListener).run(0xc000a9ff00)
porter_1       |        /go/pkg/mod/k8s.io/client-go@v0.18.8/tools/cache/shared_informer.go:738 +0x95
porter_1       | k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc0004956b0, 0xc0020d1580)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:73 +0x51
porter_1       | created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:71 +0x65
porter_1       | panic: send on closed channel [recovered]
porter_1       |        panic: send on closed channel
porter_1       | 
porter_1       | goroutine 2814 [running]:
porter_1       | k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/runtime/runtime.go:55 +0x10c
porter_1       | panic(0x237c1e0, 0x29fd1a0)
porter_1       |        /usr/local/go/src/runtime/panic.go:969 +0x1b9
porter_1       | github.com/porter-dev/porter/internal/kubernetes.(*Agent).StreamControllerStatus.func3(0x2696880, 0xc006c0af00)
porter_1       |        /porter/internal/kubernetes/agent.go:592 +0x13d
porter_1       | k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
porter_1       |        /go/pkg/mod/k8s.io/client-go@v0.18.8/tools/cache/controller.go:218
porter_1       | k8s.io/client-go/tools/cache.(*processorListener).run.func1()
porter_1       |        /go/pkg/mod/k8s.io/client-go@v0.18.8/tools/cache/shared_informer.go:744 +0xc2
porter_1       | k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc001b8c760)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:155 +0x5f
porter_1       | k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc003a0df60, 0x2a38ec0, 0xc000951200, 0x231f801, 0xc00384c1e0)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:156 +0xad
porter_1       | k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc001b8c760, 0x3b9aca00, 0x0, 0x474601, 0xc00384c1e0)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:133 +0x98
porter_1       | k8s.io/apimachinery/pkg/util/wait.Until(...)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:90
porter_1       | k8s.io/client-go/tools/cache.(*processorListener).run(0xc000a9ff00)
porter_1       |        /go/pkg/mod/k8s.io/client-go@v0.18.8/tools/cache/shared_informer.go:738 +0x95
porter_1       | k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc0004956b0, 0xc0020d1580)
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:73 +0x51
porter_1       | created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
porter_1       |        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:71 +0x65
```

- The "Last Deployed" time should be updated on a new revision when viewing the list of charts, and the dashboard should re-order the applications when they're updated.

File: internal/kubernetes/agent.go [hunk 1]
```diff
@@ -556,6 +561,8 @@ func (a *Agent) StreamControllerStatus(conn *websocket.Conn, kind string, select
 		informer = factory.Core().V1().Namespaces().Informer()
 	case "pod":
 		informer = factory.Core().V1().Pods().Informer()
+	case "secrets":
```
Comment (abelanger5) [inline internal/kubernetes/agent.go:564]: I think this line should be removed, right? We definitely don't want the option for streaming all created/updated secrets to the client.

Comment (jnfrati) [reply internal/kubernetes/agent.go:564]: Mmm, I was using it before for the chart list, but I'll migrate that section to use the helm_chart ws instead!

File: internal/kubernetes/agent.go [hunk 2]
```diff
@@ -600,6 +607,213 @@ func (a *Agent) StreamControllerStatus(conn *websocket.Conn, kind string, select
...
+	)
+
+	informer := factory.Core().V1().Secrets().Informer()
+
+	stopper := make(chan struct{})
+	errorchan := make(chan error)
+	defer close(errorchan)
+
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj, newObj interface{}) {
```
Comment (abelanger5) [inline internal/kubernetes/agent.go:695]: All three of the `UpdateFunc`, `AddFunc` and `DeleteFunc` contain a lot of repeated code. Could you add a function that just parses the the `obj` into a release object?

**Comment by jusrhee** [comment]
Added image pause screen before GHA calls webhook + blocked redeploying on other tabs

File: internal/kubernetes/agent.go (hunk 2 above)
Comment (abelanger5) [reply internal/kubernetes/agent.go:717]: Can't this be resolved in one function still?

```go
helm_object, isNotHelmRelease, err := parseSecretToHelmRelease(*secretObj, chartList)

			if isNotHelmRelease && err == nil {
				return
			}

			if err != nil {
				errorchan <- err
				return
			}

			msg := Message{
				EventType: "DELETE",
				Object:    helm_object,
			}

			if writeErr := conn.WriteJSON(msg); writeErr != nil {
				errorchan <- writeErr
				return
			}
```

Just pass the `conn` object, `secretObj`, and `chartList` to a function. Should work for `UpdateFunc`, `AddFunc` and `DeleteFunc` right?

Comment (abelanger5) [reply internal/kubernetes/agent.go:717]: Actually never mind, it's fine -- it'll get messy with the number of parameters that need to get passed through.

**Review by abelanger5 — APPROVED** [review]

**Pull request merged**